src/no-random-v4-uuid.ts

```diff
@@ -11,78 +11,108 @@ import getDocumentationUrl from './get-documentation-url.ts';
-    node.callee.object.name === uuidDefaultAlias &&
+    node.callee.object.name === aliases.uuidDefaultAlias &&
     node.callee.property.type === AST_NODE_TYPES.Identifier &&
     node.callee.property.name === 'v4');
 
-const rule: TSESLint.RuleModule<typeof NO_RANDOM_V4_UUID> = createRule({
+const isCryptoRandomUUIDCall = (node: TSESTree.CallExpression, alias?: string): boolean =>
+  (node.callee.type === AST_NODE_TYPES.Identifier && node.callee.name === alias) ||
+  (node.callee.type === AST_NODE_TYPES.MemberExpression &&
+    node.callee.object.type === AST_NODE_TYPES.Identifier &&
+    node.callee.object.name === 'crypto' &&
+    node.callee.property.type === AST_NODE_TYPES.Identifier &&
+    node.callee.property.name === 'randomUUID');
+
+const rule: TSESLint.RuleModule<typeof NO_RANDOM_V4_UUID | typeof NO_UUID_MODULE_FOR_V4> = createRule({
   name: ruleId,
   meta: {
     type: 'problem',
     docs: {
-      description: 'Disallow the use of `uuid.v4` for generating random v4 UUIDs',
+      description:
+        'Disallow the use of `uuid.v4` and `crypto.randomUUID` for generating random v4 UUIDs, and suggest replacing `uuid` module usage with `crypto.randomUUID`.',
     },
     schema: [],
     messages: {
-      [NO_RANDOM_V4_UUID]: 'Avoid using `uuid.v4` for generating random v4 UUIDs.',
+      [NO_RANDOM_V4_UUID]: 'Avoid using `crypto.randomUUID` for generating random v4 UUIDs.',
+      [NO_UUID_MODULE_FOR_V4]: 'Avoid using the `uuid` module for v4 UUID generation. Use `crypto.randomUUID` instead.',
     },
   },
   defaultOptions: [],
   create(context) {
-    let uuid4Alias: string | undefined;
-    let uuidDefaultAlias: string | undefined;
-    let hasUuidImport = false;
+    const aliases: { uuid4Alias?: string; uuidDefaultAlias?: string; cryptoRandomUUIDAlias?: string } = {};
+    let uuidImportNode: TSESTree.ImportDeclaration | null = null;
+    let hasReportedCallExpression = false;
 
     return {
       ImportDeclaration(node: TSESTree.ImportDeclaration) {
+        processImportDeclaration(node, aliases);
         if (node.source.value === 'uuid') {
-          hasUuidImport = true;
-          const result = processImportDeclaration(node, uuid4Alias, uuidDefaultAlias);
-          uuid4Alias = result.uuid4Alias;
-          uuidDefaultAlias = result.uuidDefaultAlias;
+          uuidImportNode = node;
         }
       },
       CallExpression(node: TSESTree.CallExpression) {
-        if (hasUuidImport && isUuid4Call(node, uuid4Alias, uuidDefaultAlias)) {
+        if (isUuid4Call(node, aliases)) {
+          context.report({
+            node,
+            messageId: NO_UUID_MODULE_FOR_V4,
+          });
+          hasReportedCallExpression = true;
+        } else if (isCryptoRandomUUIDCall(node, aliases.cryptoRandomUUIDAlias)) {
           context.report({
             node,
             messageId: NO_RANDOM_V4_UUID,
           });
+          hasReportedCallExpression = true;
+        }
+      },
+      'Program:exit'() {
```

Comment:
do we have to report error on import level, or it's good enought to report on call expression level?